watch.go

```diff
@@ -13,12 +13,18 @@ import (
 func (e *Executor) watchTasks(args ...string) error {
 	e.printfln("task: Started watching for tasks: %s", strings.Join(args, ", "))
 
-	// run tasks on init
+	taskCancellers := map[string]context.CancelFunc{}
+
 	for _, a := range args {
-		if err := e.RunTask(context.Background(), Call{Task: a}); err != nil {
-			e.println(err)
-			break
-		}
+		// Use of goroutines means task order is not guaranteed.
+		// Not sure if task order is a requirement. (jackmordaunt)
+		go func(a string) {
+			ctx, cancel := context.WithCancel(context.Background())
+			taskCancellers[a] = cancel
```

Comment:
After some thought... Could we also actually simplify this just use the same context (instead of taskCancellers map), and cancel all tasks through the same cancel function?